src/libsyncengine/jobs/network/upload_session/uploadsession.cpp

```diff
@@ -453,16 +456,15 @@ bool UploadSession::cancelSession() {
     }
 
     // Cancel all ongoing chunk jobs
-    _mutex.lock();
-    std::unordered_map<UniqueId, std::shared_ptr<UploadSessionChunkJob>>::iterator it = _ongoingChunkJobs.begin();
-    for (; it != _ongoingChunkJobs.end(); it++) {
-        if (it->second->sessionToken() == _sessionToken) {
-            it->second->abort();
+    for (auto &[jobId, job] : _ongoingChunkJobs) {
+        if (job.get() && job->sessionToken() == _sessionToken) {
+            LOG_INFO(_logger, "Aborting chunk job " << jobId);
+            job->abort();
         }
     }
```

Comment:
No mutex here? In the meantime, some items could have been removed from the map in `uploadChunkCallback`

Reply:
Done